airflow/sentry.py

```diff
@@ -0,0 +1,169 @@
+        """
+        Initialize the Sentry SDK.
+        """
+        ignore_logger("airflow.task")
+        ignore_logger("airflow.jobs.backfill_job.BackfillJob")
+        executor_name = conf.get("core", "EXECUTOR")
+
+        sentry_flask = FlaskIntegration()
+
+        # LoggingIntegration is set by default.
+        integrations = [sentry_flask]
+
+        if executor_name == "CeleryExecutor":
+            from sentry_sdk.integrations.celery import CeleryIntegration
+
+            sentry_celery = CeleryIntegration()
+            integrations.append(sentry_celery)
+
+        dsn = conf.get("sentry", "sentry_dsn")
+        if dsn:
+            init(dsn=dsn, integrations=integrations)
+        else:
+            # Setting up Sentry using environment variables.
+            log.debug("Defaulting to SENTRY_DSN in environment.")
+            init(integrations=integrations)
+
+    def add_tagging(self, task_instance):
+        """
+        Function to add tagging for a task_instance.
+        """
+        task = task_instance.task
+
+        with configure_scope() as scope:
+            for tag_name in self.SCOPE_TAGS:
+                attribute = getattr(task_instance, tag_name)
+                if tag_name == "operator":
+                    attribute = task.__class__.__name__
+                scope.set_tag(tag_name, attribute)
+
+    @provide_session
+    def add_breadcrumbs(self, task_instance, session=None):
+        """
+        Function to add breadcrumbs inside of a task_instance.
+        """
+        if session is None:
+            return
+        execution_date = task_instance.execution_date
+        task = task_instance.task
+        dag = task.dag
+        task_instances = dag.get_task_instances(
+            state={State.SUCCESS, State.FAILED},
+            end_date=execution_date,
+            start_date=execution_date,
+            session=session,
+        )
+
+        for ti in task_instances:
+            data = {}
+            for crumb_tag in self.SCOPE_CRUMBS:
+                data[crumb_tag] = getattr(ti, crumb_tag)
+
+            add_breadcrumb(category="completed_tasks", data=data, level="info")
+
+    def format_run_task(self, func):
+        """
+        Wrap TaskInstance._run_raw_task to support task specific tags and breadcrumbs.
+        """
+
+        @wraps(func)
+        def wrapper(task_instance, *args, session=None, **kwargs):
+            # Wrapping the _run_raw_task function with push_scope to contain
+            # tags and breadcrumbs to a specific Task Instance
+            with push_scope():
+                try:
+                    return func(task_instance, *args, session=session, **kwargs)
+                except Exception as e:
+                    self.add_tagging(task_instance)
+                    self.add_breadcrumbs(task_instance, session=session)
+                    capture_exception(e)
+                    raise
+
+        return wrapper
+
+
+Sentry = DummySentry  # type: Any
```

Comment:
If you make 

```python
class ConfiguredSentry(DummySentry):
```

then I think this could become

```python
Sentry = DummySentry  # type: DummySentry
```

Reply:
Done!